src/cli.rs

```diff
@@ -929,6 +929,7 @@ impl Opts {
                 .await
                 .context("Could not convert pin to npins format")?;
 
+            // QUESTION calling pin.update() here is surprising to me, shouldn't we try to reuse the hash from the flake in pin.try_to_pin() and only fetch here?
```

Comment:
This could've perhaps been an issue instead, but it's something i've run into it when writing the `try_to_pin()` impl for Tarball and it surprised me.

Reply:
I don't know about import-flake, but when I wrote import-niv their hashing format was too much work to fully parse to extract the detailed version information, or maybe even incompatible in some areas (I forgot). Either way, focusing only on the pin inputs at the cost of an update was a reasonable compromise